### PR TITLE
Warn when downscaling decimal columns

### LINKED ISSUE #8484: [FEA] Add WARN on truncate vs round behavior when calling astype on Decimal columns
When calling `astype` to downcast a Decimal64 column to a Decimal64 column of lower precision, trailing digits are truncated.

This is different behavior from Apache Spark and Apache Hive where `cast` operations round trailing digits, leading to confusion.

We should emit a WARN when downcasting Decimals, suggesting to combine the `round` operator with `astype` if rounding is desired.

### PR DESCRIPTION
**Pull request by ChrisJar**
Closes #8484

### CODE REVIEW

**Review by charlesbluca — APPROVED** [review]

File: python/cudf/cudf/core/column/decimal.py [hunk 1]
```diff
@@ -146,6 +147,16 @@ def _decimal_quantile(
     def as_decimal_column(
         self, dtype: Dtype, **kwargs
     ) -> "cudf.core.column.DecimalColumn":
+        if (
+            isinstance(dtype, Decimal64Dtype)
+            and isinstance(self.dtype, Decimal64Dtype)
```
Comment (shwina) [inline python/cudf/cudf/core/column/decimal.py:152]: Can we avoid this check, given. `self` is a `DecimalColumn`?

**Review by shwina — APPROVED** [review]

**Comment by shwina** [comment]
@gpucibot merge

**Comment by charlesbluca** [comment]
Any reason this never got merged?

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**